src/py/flwr/server/superlink/state/sqlite_state.py

```diff
@@ -575,6 +589,54 @@ def create_run(self, fab_id: str, fab_version: str) -> int:
         log(ERROR, "Unexpected run creation failure.")
         return 0
 
+    def store_server_public_private_key(
+        self, public_key: bytes, private_key: bytes
+    ) -> None:
+        """Store `server_public_key` and `server_private_key` in state."""
+        query = (
+            "INSERT OR REPLACE INTO credential (public_key, private_key) "
+            "VALUES (:public_key, :private_key)"
```

Comment:
Wouldn't it be an error if the one of these was already set? Or do we want to support server restarts that replace what was saved before?

Reply:
Changed the implementation to reject new keys if it's already set, and added test to ensure that.